static/app/views/performance/vitalDetail/vitalInfo.tsx

```diff
@@ -1,45 +1,105 @@
+import React from 'react';
 import {Location} from 'history';
 
+import {Organization} from 'sentry/types';
+import EventView from 'sentry/utils/discover/eventView';
 import {WebVital} from 'sentry/utils/discover/fields';
+import MetricsRequest from 'sentry/utils/metrics/metricsRequest';
 import VitalsCardDiscoverQuery from 'sentry/utils/performance/vitals/vitalsCardsDiscoverQuery';
+import {decodeScalar} from 'sentry/utils/queryString';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import useApi from 'sentry/utils/useApi';
 
 import {VitalBar} from '../landing/vitalsCards';
+import {getVitalData} from '../landing/widgets/widgets/vitalWidgetMetrics';
 
-type Props = {
+import {vitalToMetricsField} from './utils';
+
+type ViewProps = Pick<
+  EventView,
+  'environment' | 'project' | 'start' | 'end' | 'statsPeriod'
+>;
+
+type Props = ViewProps & {
   location: Location;
   vital: WebVital | WebVital[];
+  orgSlug: Organization['slug'];
   hideBar?: boolean;
   hideStates?: boolean;
   hideVitalPercentNames?: boolean;
   hideDurationDetail?: boolean;
+  isMetricsData?: boolean;
 };
 
-function VitalInfo(props: Props) {
-  const {
-    vital,
-    location,
-    hideBar,
-    hideStates,
-    hideVitalPercentNames,
-    hideDurationDetail,
-  } = props;
+function VitalInfo({
+  orgSlug,
+  start,
+  end,
+  statsPeriod,
+  project: projectIds,
+  environment,
+  vital,
+  location,
+  hideBar,
+  hideStates,
+  hideVitalPercentNames,
+  hideDurationDetail,
+  isMetricsData,
+}: Props) {
+  const api = useApi();
+
+  const vitals = Array.isArray(vital) ? vital : [vital];
+
+  function renderContent({
```

Comment:
This smells like it should either be inlined or split into a different component. I would say it's probably an antipattern to define renderContent fn for functional components and we should instead try to keep the render logic as concise as possible :) 
```
{({isLoading, vitalsData}) => isMetricData <Component1/> : <Component2/>
```

Reply:
sure thing... I've split this component in one of my open PRs but after discussing with the performance team, we decided to no fork widgets/components anymore as it can be hard to do maintenance, so as the content is basically the same, I thought that a reusable function was be a good option in this case

Reply:
I agree with Jonas's point here, you making a new component definition on the fly basically, every time this FC gets called. You might be hitting some extra re-renders here.